deploy-tool/main.go

```diff
@@ -74,6 +81,32 @@ type SingletonIngestor struct {
 	BatchSigningKeyName       string `json:"batch_signing_key_name"`
 }
 
+type ManifestStorage interface {
```

Comment:
It seems like we only have this and interface `ManifestWriter` anymore so that `deploy-tool/main_test.go` can provide its own `FakeManifestStorage` that implements the interface, but it seems like we could delete this if `deploy-tool/main_test.go` could use `memDS` from `key-rotator/storage/manifest_test.go`. Then again, the objective is to delete `deploy-tool` so it's probably not worth the hassle of sharing test code.

Reply:
I made this change based on the principles of "define interfaces at point-of-use, not point-of-implementation" &  "take interfaces, return concrete types". But I had the same testing reservations that you do -- it'd be nice if we could centralize a single "for-tests" Manifest implementation. And then I noticed [Effective Go](https://golang.org/doc/effective_go#generality) suggests that sometimes it _is_ OK to return an interface (and the stdlib certainly does in some cases).

OTOH, out-of-package tests probably want a fake that lets them see what manifests are read & written, rather than what bytes are read & written. But storage.Manifest w/ memDS works at the level of bytes (because I really do want that level of control/testing when testing Manifest in-package).

I settled on making the struct Manifest private, introducing a new Manifest interface with the expected methods, and then adding a storage/test package with a Manifest fake inspired by deploy-tool's FakeManifestStorage to show that this layout works for a centralized testing implementation. I think this is worthwhile because I'll reuse it for the key-rotator main logic.